### PR TITLE
[ROCM] Fixed `tt.atomic_rmw` for f16 type.

### PR DESCRIPTION
**Pull request by dfukalov**
Fixed bug when `AtomicRMWOp` fails to process packed f16 operands.

### CODE REVIEW

File: lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 1]
```diff
@@ -658,7 +658,9 @@ struct AtomicRMWOpConversion
                 icmp_slt(mul(tid, i32_val(elemsPerThread)), i32_val(numElems)));
 
     auto vecTy = vec_ty(valueElemTy, vec);
+    auto retType = vec == 1 ? valueElemTy : vecTy;
     SmallVector<Value> resultVals(elemsPerThread);
+    const bool f16v2 = vec == 2 && valueElemTy.isF16();
     for (size_t i = 0; i < elemsPerThread; i += vec) {
       Value rmwVal = undef(vecTy);
       for (int ii = 0; ii < vec; ++ii) {
```
Comment (binarman) [inline lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp:666]: Looks like this loop was added to handle case with f16.
Maybe delete it?

Comment (dfukalov) [reply lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp]: Yes, thanks, fixed.

**Comment by dfukalov** [comment]
> Thanks @dfukalov Is there a test that this fixes?
Yes, sure, it is in the patch, added to `test/Conversion/tritongpu_to_llvm.mlir`

**Review by jayfurmanek — APPROVED** [review]
This looks good! Thanks!

**Pull request merged**